ext/zlib/zlib.c

```diff
@@ -1164,7 +1164,21 @@ PHP_FUNCTION(deflate_add)
 	ctx->avail_in = in_len;
 	ctx->avail_out = ZSTR_LEN(out);
 
-	status = deflate(ctx, flush_type);
+	buffer_used = 0;
+
+	do {
+		if (ctx->avail_out == 0) {
+			/* more output buffer space needed; realloc and try again */
+			/* adding 64 more bytes solved every issue I have seen    */
+			/* the + 1 is for the string terminator added below */
+			out = zend_string_realloc(out, ZSTR_LEN(out) + 64 + 1, 0);
```

Comment:
Oh. Makes sense, though it doesn't make any difference. I already pushed it now; doesn't really matter.